### DISCUSSION TITLE
Name change of label through UI removes label from PRs/Issues

### DISCUSSION THREAD

**Discussion by ces131** [crazy-max/ghaction-github-labeler#196] [Q&A] [answered]
Hi, I'm not sure how possible this is -

I have the action running when the list of labels is changed or a label is created, deleted, or updated through the Github UI. If a user changes the name of a label through the UI, the action will delete that label and create a new one with the original name. The problem with this is that the renamed label will be removed from all issues/PRs. Ideally, you would want just the name to change and sort of follow the `from_name:` process, keeping the label on any issues/PRs it's placed on. However, this seems to be super difficult to detect what the label was changed from when updated through the UI so I'm not sure how possible this even is with an action.

If adding it to the action doesn't seem worthwhile/deems too difficult with what's there to work with I was wondering if you had any suggestions to overcome this? I was thinking I could trigger a different Github action if the label is updated through the UI versus when one is added/deleted. Thanks!

**Comment by crazy-max** [comment, marked as answer]
Hi @ces131,

This action only handles labels as code. Therefore you should not modify/add/remove labels from the webui and do not use label events to trigger this action as this could interfere with the definition of labels included in `labels.yml`.
This action is useful if you want to initialize a repo with predefined labels. Once you are satisfied, you can disable this action with the input `dry_run: true` and let the user modify current labels as he wishes.

**Comment by ces131** [comment]
Gotcha @crazy-max, we were using this action mainly to standardize/control labels in a really large repo, making the `labels.yml` file the source of truth and having it revert changes made through the UI. Your action is super helpful in this case so we will probably be using it alongside another action for specifically updating a label. Thanks for the explanation! :) 

I'll go ahead and close this issue. Thanks again!
